### ISSUE TITLE
Subscription Auto Refund, finishTransaction Error 

### ISSUE THREAD

**Issue by kivancagaogluu** [dooboolab-community/react-native-iap#2593]
**Description**

I am using same code within examples folder. My app is in production and every subscription refunded after 3 days by Google without customers knowledge.  I have tried both emulator and android device and got a log which does not give that much information.

**Environment:**

- react-native-iap: 12.10.7
- react-native0.72.4:
- Platforms (iOS, Android, emulator, simulator, device): Android Device & Emulator

Example Code

```javascript
useEffect(() => {
    const checkCurrentPurchase = async () => {

      try {
        if (currentPurchase?.productId) {
          await finishTransaction({
            purchase: currentPurchase,
            isConsumable: true,
          });

          setOwnedSubscriptions((prev) => [
            ...prev,
            currentPurchase?.productId,
          ]);
        }
      } catch (error) {
        console.log("error", error);
        if (error instanceof PurchaseError) {
          errorLog({message: `[${error.code}]: ${error.message}`, error});
        } else {
          errorLog({message: 'handleBuyProduct', error});
        }
      }
    };

```

Output
```
 ERROR  An error happened handleBuyProduct [Error: Google is indicating that we have some issue connecting to payment.]
 ```

Does anyone know solution or root of the problem? Any help will be highly appreciated

**Comment by kivancagaogluu** [comment]
I still do my tests but changing isConsumable looks like the fix i have been looking for. @antoine-souesme

**Comment by yhostc** [comment]
@kivancagaogluu Has a good solution been found for this problem?

**Comment by kivancagaogluu** [comment]
As i mentioned previous comments; i changed isConsumable value to false which is make sense because user can not buy two times a subscription.

**Comment by yhostc** [comment]
Yes, I have set isConsumable to false. This exception prompt is no longer available on the real machine, but I still receive this exception on the simulator. This makes me worry about what the results will be on more machines.

**Comment by yhostc** [comment]
I've found the right way to handle this problem. For subscription products, Google Play states in the documentation that the acknowledge interface needs to be called immediately after the server grants the rights to confirm the transaction.

https://developer.android.com/google/play/billing/security#verify

**Comment by kivancagaogluu** [comment]
I was unable to try subscription on emulator and i sended an email to Google Support. I dont know what they did but now it works on emulator too.

**Comment by darioduarte** [comment]
> I still do my tests but changing isConsumable looks like the fix i have been looking for. @antoine-souesme

could you mentioned the soluction? I cannot found this

**Comment by saivarma49** [comment]
i have subscribed successfully but currentPurchase showing undefined

**Comment by hyochan** [comment]
I’m closing all issues reported in versions below 14, as the library now supports the new architecture with NitroModules and has been completely revamped.
